### PR TITLE
DetectionMode adjustments

### PR DESCRIPTION
**Pull request by tschulze**
## Checklist

* [X] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [X] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [X] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [X] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [X] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [X] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Models/DetectionMode.swift [hunk 1]
```diff
@@ -28,8 +28,8 @@ enum DetectionMode {
 }
 
 extension DetectionMode {
-	static func from(backgroundStatus: UIBackgroundRefreshStatus) -> DetectionMode {
-		switch backgroundStatus {
+	static func fromBackgroundRefreshStatus() -> DetectionMode {
```
Comment (lennartstolz) [inline src/xcode/ENA/ENA/Source/Models/DetectionMode.swift:31]: IMHO this change would make the method harder to test, as you would need to swizzle/mock the `backgroundRefreshStatus` of `UIApplication.shared` when you'd like to test if it returns the expected values.

What do you think about moving it into the method signature as a default argument?

```swift
static func from(backgroundStatus: UIBackgroundRefreshStatus = UIApplication.shared.backgroundRefreshStatus) { ... }
```

This way you would get rid of the duplicated code you wanted to remove, but you could even add tests for this method.

Comment (tschulze) [reply src/xcode/ENA/ENA/Source/Models/DetectionMode.swift:31]: This was my first idea, but `DetectionMode.from()` looks a little bit strange.

Comment (lennartstolz) [reply src/xcode/ENA/ENA/Source/Models/DetectionMode.swift]: Fair point, didn't think about that. But what's about?

```swift
static func fromBackgroundStatus(_ backgroundStatus: UIBackgroundRefreshStatus = UIApplication.shared.backgroundRefreshStatus) { ... }
```

This should solve it, no? :)

Comment (tschulze) [reply src/xcode/ENA/ENA/Source/Models/DetectionMode.swift]: I kept the old method (for mocking) and the new one. Does this look better for you?

**Review by ChristianKienle — APPROVED** [review]

**Pull request merged**